### PR TITLE
contracts-bedrock: loadAllocs support

### PR DESCRIPTION
**Pull request by tynes**
**Description**

Small tweaks to the L2 contracts tests to make the
new method of setting up the L2 state work. This depends
on the latest build of `ci-builder` and updating the
foundry version in https://github.com/ethereum-optimism/optimism/pull/8144.

### CODE REVIEW

**Comment by tynes** [comment]
We are expecting these tests to fail until we have the new `ci-builder` image running in CI

File: packages/contracts-bedrock/test/SequencerFeeVault.t.sol [hunk 1]
```diff
@@ -38,6 +39,7 @@ contract SequencerFeeVault_Test is CommonTest {
     function test_receive_succeeds() external {
         uint256 balance = address(sequencerFeeVault).balance;
 
+        vm.deal(alice, 100);
```
Comment (tynes) [inline packages/contracts-bedrock/test/SequencerFeeVault.t.sol]: this is prob dead code now

File: packages/contracts-bedrock/test/setup/Setup.sol [hunk 2]
```diff
@@ -28,6 +28,7 @@ import { L1StandardBridge } from "src/L1/L1StandardBridge.sol";
 import { AddressManager } from "src/legacy/AddressManager.sol";
 import { L1ERC721Bridge } from "src/L1/L1ERC721Bridge.sol";
 import { AddressAliasHelper } from "src/vendor/AddressAliasHelper.sol";
+import { SafeCall } from "src/libraries/SafeCall.sol";
```
Comment (tynes) [inline packages/contracts-bedrock/test/setup/Setup.sol]: this is dead code now

**Comment by tynes** [comment]
This error doesn't seem to happen locally, need to figure out why it is failing in CI still

**Review by mslipper — APPROVED** [review]

File: .circleci/config.yml [hunk 3]
```diff
@@ -378,7 +378,7 @@ jobs:
   contracts-bedrock-checks:
     docker:
       - image: <<pipeline.parameters.ci_builder_image>>
-    resource_class: medium
+    resource_class: xlarge
```
Comment (trianglesphere) [inline .circleci/config.yml:381]: I'm not sure that the checks needs an xl. The tests seem more likely to need it, but I'm still not super clear on why

Comment (tynes) [reply .circleci/config.yml:381]: The checks fail without an xl, see [08d3fa2](https://github.com/ethereum-optimism/optimism/pull/8148/commits/08d3fa258d829d6ea5368e7e2b938eb46b937ce0)

Comment (trianglesphere) [reply .circleci/config.yml:381]: Oh, I got confused b/c I saw that they gas snapshot step had failed, but I thought it failed for the reason that was fixed later on

**Review by clabby — APPROVED** [review]

File: .circleci/config.yml (hunk 3 above)
Comment (tynes) [reply .circleci/config.yml:381]: all good

**Pull request merged**